Add CI that builds CloudFlare Pages previews on forks

johndmulhausen opened the pull request:
CloudFlare Pages integration doesn't run on forks. But, because the forks are public, we can do custom CI that manually `git clone`s those forks, and, after checking them out, runs the CloudFlare Pages build for us, commenting back in the originating PR with a link.

See #1206 for an example of this CI in action

johndmulhausen committed:
Add CI that builds CloudFlare Pages previews on forks

CloudFlare Pages integration doesn't run on forks. But, because the forks are public, we can do custom CI that manually `git clone`s those forks, and, after checking them out, runs the CloudFlare Pages build for us, commenting back in the originating PR with a link.

johndmulhausen committed:
Linkback to logs, error handling

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update quickstart.md

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
Update forks-build-on-cloudflare.yml

johndmulhausen committed:
revert quickstart.md to main

ngrayluna commented on .github/workflows/forks-build-on-cloudflare.yml:
```diff
@@ -0,0 +1,228 @@
+          fi
+      
+      # Debug directory structure
+      - name: Debug directory structure
```
What happens w/out these checks? Curious what errors you were hitting.

ngrayluna reviewed (approved):
Left a question (for my understanding). Otherwise looks good.

johndmulhausen committed:
Minor improvements to Quickstart (#1206)

Mission 1 of this PR: To improve some things that are currently a bit
awkward in the quickstart.
Mission 2: Testing new functionality that deploys CloudFlare preview
builds on PRs created from forks. (See the GitHub Actions
comment/CloudFlare links below)

Mission 2 is why this PR targets the `forks-build-on-cloudflare` branch
(so you can see the CI in action), and is also why the correct merge
order is: this PR first, then #1205 after to finally publish into main.
